cirq-core/cirq/qis/measures_test.py

```diff
@@ -164,6 +166,77 @@ def test_fidelity_product_states():
         )
 
 
+def test_fidelity_different_simulator_outputs_compatible():
+    from cirq.sim.act_on_state_vector_args import _BufferedStateVector
+    from cirq.sim.act_on_density_matrix_args import _BufferedDensityMatrix
+
+    gate_domain = {
+        cirq.CNOT: 2,
+        cirq.CZ: 2,
+        cirq.H: 1,
+        cirq.S: 1,
+        cirq.SWAP: 2,
+        cirq.X: 1,
+        cirq.Y: 1,
+        cirq.Z: 1,
+    }
+    circuits = [
+        cirq.testing.random_circuit(qubits=5, n_moments=30, op_density=1, gate_domain=gate_domain)
+        for _ in range(2)
+    ]
+
+    def final_state(sim: cirq.SimulatorBase):
+        return [
+            sim.simulate(c)._final_step_result_typed._sim_state.create_merged_state()._state
```

Comment:
@95-martin-orion One thing to note here is that all of these states, regardless of the simulator, are found in the same (obscure) place. I think we should make this public after renaming ActOnArgs to SimulationState.